### PR TITLE
Add refactoring around replicas change handling

### PR DESCRIPTION
**Pull request by fvaleri**
This puts all replicas change logic inside the checkReplicasChange() method. It also improves results handling inside updateInternal() so that results is not passed deep down the call stack.

This is also tested with the ST currently under review.

### CODE REVIEW

**Review by scholzj — APPROVED** [review]
LGTM assuming it works.

File: topic-operator/src/test/java/io/strimzi/operator/topic/v2/BatchingTopicControllerTest.java [hunk 1]
```diff
@@ -309,185 +327,49 @@ public void replicasChangeShouldMoveToPendingFromNew(boolean cruiseControlEnable
...
         } else {
-            assertThat(results.keySet(), empty());
+            assertThat(results.errors().count(), is(1L));
+            assertThat(results.errors().findFirst().get().getValue(), instanceOf(TopicOperatorException.NotSupported.class));
         }
     }
 
-    @ParameterizedTest
-    @ValueSource(booleans = { true, false })
-    public void replicasChangeShouldMoveToOngoingFromPending(boolean cruiseControlEnabled) {
```
Comment (tombentley) [inline topic-operator/src/test/java/io/strimzi/operator/topic/v2/BatchingTopicControllerTest.java:331]: Stupid question maybe, but why do we no longer need to test this?

Comment (fvaleri) [reply topic-operator/src/test/java/io/strimzi/operator/topic/v2/BatchingTopicControllerTest.java:331]: This is covered by replicasChangeShouldMoveFromNewToPending. It doesn't really matter which state transition, as I'm mocking the resulting status (i.e. replicasChangeHandler behavior). What I'm really testing is the controller logic around the replicasChangeHandler invocation, which leads to results.ok() or results.error(). Should we use a better name for this test?

**Review by tombentley — APPROVED** [review]
Thanks @fvaleri !

**Pull request merged**